### PR TITLE
[google_sign_in] Added NonNull annotations, reduce Guava usage

### PR DESCRIPTION
**Pull request by ened**

### CODE REVIEW

**Comment by JamesMcIntosh** [comment]
This upgrade of dependencies when used in conjunction with the other new firebase plugin versions causes problems due to the transitive `android.arch.core version 1.1.0` dependencies from `play-services-auth`.
When mixing the 1.1.0 and 1.1.1 versions the dex fails as follows:
```
Gradle build fails while trying to dex with "Program type already present: android.arch.core.util.Function"
```
see: https://stackoverflow.com/questions/50524672/build-fails-with-program-type-already-present-android-arch-core-util-function

As a workaround until 'play-services-auth' bumps it's dependency I recommend either explicitly specifying version 1.1.1 of these libraries as a dependency or adding a note in the README documentation.
```
implementation "android.arch.core:runtime:1.1.1"
implementation "android.arch.core:common:1.1.1"
```

**Comment by ened** [comment]
@JamesMcIntosh Would adding:

```
implementation "android.arch.core:runtime:1.1.1"
implementation "android.arch.core:common:1.1.1"
```

To this components build.gradle help? If so I will update the PR.

**Comment by JamesMcIntosh** [comment]
In my project I have declared it as a dependency, I think it specifying it in this project should be enough to bring it all inline with the other firebase projects, I'll test it tomorrow morning and let you know.

**Comment by JamesMcIntosh** [comment]
I tested and adding them as api dependencies fixes the issue.

```
api "android.arch.core:runtime:1.1.1"
api "android.arch.core:common:1.1.1"
```

I'm feel more like this should be added to the README as a known issue for this release, suggested workaround as this is a problem of interop between libraries, but since they are both libraries un the same project maybe one of the Google crew would like to throw in their thoughts. @bparrishMines ?

I dug a little bit further to see where the dependency incompatibility came from:

1. `cloud_firestore` bumps the version of `firebase`
2. This in turn bumps the version of `android.arch.lifecycle` to 1.1.1 
3. This bumps the transitive dependency of `android.arch.lifecycle` in `google_sign_in`
4. It falls apart because `android.arch.core:runtime` is not a dependency of `android.arch.lifecycle` so remains at 1.1.0.
 
```
+--- project :google_sign_in
|    \--- com.google.android.gms:play-services-auth:16.0.1
|         +--- com.google.android.gms:play-services-auth-api-phone:16.0.0
|         |    +--- com.google.android.gms:play-services-base:16.0.1
|         |    |    +--- com.google.android.gms:play-services-basement:16.0.1
|         |    |    |    \--- com.android.support:support-v4:26.1.0 -> 27.1.1
|         |    |    |         +--- com.android.support:support-compat:27.1.1
|         |    |    |         |    +--- com.android.support:support-annotations:27.1.1
|         |    |    |         |    \--- android.arch.lifecycle:runtime:1.1.0 -> 1.1.1
|         |    |    |         |         +--- android.arch.lifecycle:common:1.1.1
...
+--- project :cloud_firestore
|    \--- com.google.firebase:firebase-firestore:17.1.1
|         +--- android.arch.lifecycle:common:1.1.1 (*)
|         +--- android.arch.lifecycle:runtime:1.1.1 (*)
```

**Comment by ened** [comment]
@JamesMcIntosh updated the PR to match latest status on master. This would be good for a review now as it's a lot simpler.

**Comment by JamesMcIntosh** [comment]
Looks good to me but one of the Google crew will need to accept it.

File: packages/google_sign_in/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.2.5
+
+* Add correct @NonNull annotations to Android version to reduce compiler warnings
```
Comment (cyanglaz) [inline packages/google_sign_in/CHANGELOG.md]:
```suggestion
* Update guava to `27.0.1-android`.
* Add correct @NonNull annotations to reduce compiler warnings.
```

**Comment by cyanglaz** [comment]
@ened It seems like the CI is not happy about the formatting, I think I can merge this after you fix the CI errors.

**Comment by ened** [comment]
@cyanglaz The CI is not happy about other things, too - but I don't think it's related to my PR. Please take a look.

**Comment by ened** [comment]
@cyanglaz it’s green

**Pull request merged**